db/sg_replicate_cfg.go

```diff
@@ -120,23 +124,30 @@ type ReplicationUpsertConfig struct {
 func (rc *ReplicationConfig) ValidateReplication(fromConfig bool) (err error) {
 
 	// Cancel is only supported via the REST API
-	if rc.Cancel {
-		if fromConfig {
-			return base.HTTPErrorf(http.StatusBadRequest, "cancel=true is invalid for replication in Sync Gateway configuration")
-		}
+	if rc.Cancel && fromConfig {
+		return base.HTTPErrorf(http.StatusBadRequest, "cancel=true is invalid for replication in Sync Gateway configuration")
 	}
 
 	// Adhoc is only supported via the REST API
-	if rc.Adhoc {
-		if fromConfig {
-			return base.HTTPErrorf(http.StatusBadRequest, "adhoc=true is invalid for replication in Sync Gateway configuration")
-		}
+	if rc.Adhoc && fromConfig {
+		return base.HTTPErrorf(http.StatusBadRequest, "adhoc=true is invalid for replication in Sync Gateway configuration")
 	}
 
 	if rc.Remote == "" {
 		return base.HTTPErrorf(http.StatusBadRequest, "Replication remote must be specified.")
 	}
 
+	remoteURL, err := url.Parse(rc.Remote)
+	if err != nil {
+		base.Debugf(base.KeyReplicate, "Failed to parse replication remote URL [%s]: %v", remoteURL, err)
+		return base.HTTPErrorf(http.StatusBadRequest, "Replication remote URL is invalid")
```

Comment:
We're losing error information here that was in the old `url.Parse` code. It's going to be difficult for users to determine what the problem with the URL actually was without it.

```suggestion
 		return base.HTTPErrorf(http.StatusBadRequest, "Replication remote URL [%s] is invalid: %v", remoteURL, err)
```

Reply:
I agree about returning the error information, but do we also need to redact credentials if present in remoteURL?

Reply:
Added changes to redact the credentials if present in both remote URL and the error message when returning error information.